api/src/main/java/org/apache/iceberg/events/UpdateEvent.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.events;
+
+public interface UpdateEvent {
```

Comment:
What is the value of having an interface for all types of updates with no methods?

The listener code routes these events using `listeners.get(event.getClass())` so the interface doesn't help to subscribe to all events.

Reply:
I was thinking that there may be some update operations where commit is being performed in their superclass. In those cases, we have to get the sub class' update event. Seeing through the code again, I don't see this and where a commit it being performed in a class it has all the information to send the event.  

I'll revert this change

Reply:
The issue is that I need to call the listeners after commit, but I don't have all the information to construct the event. For instance for `SnapshotProducer` where we are committing  I don't have information on how to construct the event and I'm relying on the subclass `MergingSnapshotProducer` to construct the right event properly

Reply:
Makes sense. We could also have `updateEvent()` return `Object`, since there are no requirements for objects passed to `Listeners.notifyAll`.